### DISCUSSION TITLE
Strongswan does not detect NAT when interface is not up yet

### DISCUSSION THREAD

**Discussion by robertdahlem** [strongswan/strongswan#708] [Q&A]
Hi,

this happens on a Raspi running Raspbian 11 (bullseye). Strongswan version is 5.9.1-1+deb11u1.

This Raspi boots pretty fast and starts Strongswan before the network interface is up. This leads to Strongswan NOT logging
`local host is behind NAT, sending keep alives`
and NOT switching from port 500 to 4500. In the end, the CHILD_SA gets established, but no traffic is coming through the tunnel.

I have a cron job that detects blocked tunnels, terminates the IKE_SA and initiates a new CHILD_SA. This new tunnel transports traffic then.

My workaround is to configure `encap = yes`. Would I have to expect any negative side effects from that?

Here is my [swanctl.conf](https://github.com/strongswan/strongswan/files/7469646/swanctl.conf.txt) and my [charon.log](https://github.com/strongswan/strongswan/files/7469652/client.log) with the bad case first and the good case directly after that.

Without my cron job the tunnel would be stuck in the blocked state. Should we consider this a bug?

Regards,
Robert

**Comment by robertdahlem** [comment]
This is not Raspi related. I was able to reproduce it on a fresh Debian 11 install.

**Reply by tobiasbrunner** [reply]
Note that you can edit your question to add stuff like that.

**Comment by tobiasbrunner** [comment]
Yes, that's a known issue. There is a patch in the _natd-fixes_ branch, but I'm not sure if it's the best approach. So there is also an alternative solution in the _src-ip-abort_ branch, but since `charon.retry_initiate_interval` is not enabled by default, this might be too radical.

> My workaround is to configure `encap = yes`. Would I have to expect any negative side effects from that?

If you are behind a NAT anyway, there is no difference at all. The only thing this option does is force a NAT situation on the client via a fake NAT detection notify for the source address. It's also what the first patch above does implicitly in the IPv4 case if no source IP can be determined.
